app/renderer/js/pages/preference/general-section.ts

```diff
@@ -493,4 +499,66 @@ export default class GeneralSection extends BaseSection {
 			await this.factoryResetSettings();
 		});
 	}
+
+	initSpellChecker(): void {
+		// The elctron API is a no-op on macOS and macOS default spellchecker is used.
+		if (process.platform === 'darwin') {
+			const note = document.querySelector('#note');
+			note.append(t.__('On macOS, the OS spellchecker is used.'));
+			note.append(document.createElement('br'));
+			note.append(t.__('Change the language from System Preferences → Keyboard → Text → Spelling.'));
+		} else {
+			const note = document.querySelector('#note');
+			note.append(t.__('You can select a maximum of 3 languages for spellchecking.'));
+			const spellDiv = document.querySelector('#spellcheck-langs');
+			spellDiv.innerHTML += `
+				<div class="setting-description">${t.__('Spellchecker Languages')}</div>
+				<input name='spellcheck' placeholder='Enter Languages'>`;
+
+			const availableLanguages = session.fromPartition('persist:webviewsession').availableSpellCheckerLanguages;
+			let languagePairs: Map<string, string> = new Map();
+			availableLanguages.forEach((l: string) => {
+				if (ISO6391.getName(l) !== '') {
+					languagePairs.set(ISO6391.getName(l), l);
+				}
+			});
+
+			// Manually set names for languages not available in ISO6391
+			languagePairs.set('English (AU)', 'en-AU');
+			languagePairs.set('English (CA)', 'en-CA');
+			languagePairs.set('English (GB)', 'en-GB');
+			languagePairs.set('English (US)', 'en-US');
+			languagePairs.set('Spanish (Latin America)', 'es-419');
+			languagePairs.set('Spanish (Argentina)', 'es-AR');
+			languagePairs.set('Spanish (Mexico)', 'es-MX');
+			languagePairs.set('Spanish (US)', 'es-US');
+			languagePairs.set('Portuguese (Brazil)', 'pt-BR');
+			languagePairs.set('Portuguese (Portugal)', 'pt-PT');
+			languagePairs.set('Serbo-Croatian', 'sh');
+
+			languagePairs = new Map([...languagePairs].sort(undefined));
+
+			const tagField = document.querySelector('input[name=spellcheck]');
+			const tagify = new Tagify(tagField, {
+				whitelist: [...languagePairs.keys()],
+				enforceWhitelist: true,
+				maxTags: 3,
+				dropdown: {
+					enabled: 0,
+					maxItems: Infinity,
+					closeOnSelect: false,
+					highlightFirst: true
+				}
+			});
+
+			const configuredLanguages: string[] = ConfigUtil.getConfigItem('spellcheck-languages').map((code: string) => [...languagePairs].filter(pair => (pair[1] === code))[0][0]);
```

Comment:
That’s still quadratic.

```suggestion
			const configuredLanguages: string[] =
				ConfigUtil.getConfigItem('spellcheck-languages')
					.map((code: string) => ISO6391.getName(code));
```

Reply:
We won't be able to do this because ISO6391 doesn't have all the languages , so we have to map from languagePairs.
Even though, this looks quadratic, `ConfigUtil.getConfigItem('spellcheck-languages')` would at max have 3 values, so in reality the algorithm is actually linear on the size of languagePairs, which I think is okay.

Reply:
@andersk 
```ts 
.map((code: string) => ISO6391.getName(code)); 
``` 
You're correct. This IS the convention. But lack of all language pairs makes it unusable here.
DO you have some other way in mind to make it linear ?